src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpCodeParser.cs

```diff
@@ -110,6 +110,17 @@ protected void MapDirectives(Action handler, params string[] directives)
             {
                 _directiveParsers.Add(directive, handler);
                 Keywords.Add(directive);
+
+                // These C# keywords are reserved for use in directives. It's an error to use them outside of
+                // a directive. This code removes the error generation if the directive *is* registered.
+                if (string.Equals(directive, "class", StringComparison.OrdinalIgnoreCase))
```

Comment:
This change was not a straightforward as Taylor said. Remove these as 'reserved words' unconditionally broke about 200 tests.